subsys/bluetooth/mesh/net.c

```diff
@@ -132,17 +132,30 @@ static bool msg_cache_match(struct bt_mesh_net_rx *rx,
 
 	for (i = 0U; i < ARRAY_SIZE(msg_cache); i++) {
 		if (msg_cache[i] == hash) {
+			rx->msg_cache_idx = i;
```

Comment:
This is kind of unnecessary, since in this branch we return and say a match was found, i.e. no further processing of the packet happens.